beeHealthy/src/main/java/br/com/devcapu/bee_healthy/domain/health/calculator/HealthCalculator.java

```diff
@@ -0,0 +1,35 @@
+package br.com.devcapu.bee_healthy.domain.health.calculator;
+
+import br.com.devcapu.bee_healthy.domain.health.BMI;
+import br.com.devcapu.bee_healthy.domain.health.BasalEnergyExpenditure;
+import br.com.devcapu.bee_healthy.domain.health.CaloriesToCommitObjective;
+import br.com.devcapu.bee_healthy.domain.health.Gender;
+import br.com.devcapu.bee_healthy.domain.health.Health;
+import br.com.devcapu.bee_healthy.domain.health.TotalEnergyExpenditure;
+import br.com.devcapu.bee_healthy.domain.user_info.UserInfo;
+
+public class HealthCalculator {
+
+    public Health calculate(UserInfo userInfo) {
+        BMI bmi = new BMICalculator().calculate(userInfo);
+
+        BasalEnergyExpenditureCalculator basalEnergyExpenditureCalculator = getBasalEnergyExpenditureCalculator(userInfo);
```

Comment:
Se associar o `BEECalculator` ao `gender`, ficaria algo como:

```suggestion
        BasalEnergyExpenditureCalculator basalEnergyExpenditureCalculator = userInfo.getGender().getBasalEnergyExpenditureCalculator();
```

ou, pra evitar o [trenzinho](https://hackernoon.com/object-oriented-tricks-2-law-of-demeter-4ecc9becad85), faça o `UserInfo` dizer qual calculadora deve ser usada:

```suggestion
        BasalEnergyExpenditureCalculator basalEnergyExpenditureCalculator = userInfo.getBasalEnergyExpenditureCalculator();
```

Daí, o `UserInfo` pegaria do `gender`.